queries.sql

```diff
@@ -13,3 +13,58 @@ SELECT * FROM animals WHERE neutered = TRUE;
 SELECT * FROM animals WHERE name <> 'Gabumon';
 
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+BEGIN;
+
+UPDATE animals SET species = 'unspecified';
+
+SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+
+BEGIN;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+SELECT * FROM animals;
+
+COMMIT;
+
+SELECT * FROM animals;
+
+
+BEGIN;
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT my_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO my_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+
+COMMIT;
+
+SELECT * FROM animals;
+
+
+
+SELECT COUNT(*) FROM animals;
```

Comment:
- [ ] While the queries in the image below are not here, I will also encourage you to add each transaction screenshot, I can see you did many transactions and only the last one was added.
![image](https://github.com/bohaz/Vet-clinic-database/assets/86200356/ee63b16c-4469-40e7-9595-407d2f3b92de)

Reply:
Hi @Olamarx 

- As you can see I did all the transactions and I am showing how the final table looks. 

![queryandupdate1](https://github.com/bohaz/Vet-clinic-database/assets/127757182/12136843-78a1-4bae-aed2-c439783fa9d0)
![queryandupdate2](https://github.com/bohaz/Vet-clinic-database/assets/127757182/8426db53-cbb2-4bac-9411-65c1c1e1fd77)
![queryandupdate3](https://github.com/bohaz/Vet-clinic-database/assets/127757182/fcf4da85-1682-47d7-a5fd-1af95050ac6a)
![queryandupdate4](https://github.com/bohaz/Vet-clinic-database/assets/127757182/b66813cb-5089-44fa-9d62-97dee2a2d71c)
![queryandupdate5](https://github.com/bohaz/Vet-clinic-database/assets/127757182/bed58c25-f497-4e2c-a097-610fa399a26b)
![Vet-clinic1](https://github.com/bohaz/Vet-clinic-database/assets/127757182/b7539bc1-1a57-4025-86a5-78d350a38538)
![Vet-clinic2](https://github.com/bohaz/Vet-clinic-database/assets/127757182/b899a2aa-9672-4434-95dd-bf159283e4a5)


- How is required, I'm using `ALTER` to add the new column.

![alter (2)](https://github.com/bohaz/Vet-clinic-database/assets/127757182/54fd67d0-b98d-4221-86c0-f4a532739e97)